### ISSUE TITLE
Uploads Not Working

### ISSUE THREAD

**Issue by ZY-Ang** [keichan34/wordpress-on-lambda#3]
Hey Keita, thanks for this. This is a great project and POC for wordpress on serverless!

I got everything to work on lambda except uploads. Uploads of pngs/jpg types time out at the default limit of 30s with not much error log output even after initializing `handler.php` with `ob_implicit_flush()`. Something is definitely causing uploads to hang and the buffer does not flush.

I'm debugging probable causes and I notice your php layer is utilizing `73:11` from your own account. where I am using stackery's `73:3` layer in ap-southeast-1. Could this be the issue? Or did you install any other dependencies like phpgd or ImageMagick for the s3-uploads plugin to work?

I think another reason is probably some setting that is changed directly on the [wordpress console](https://dskhgdbzphjkm.cloudfront.net/wp-admin/options.php). If you did and still remember what you changed, could you share it? It would be of great help! :)

Edit: It seems stackery already merged your PR, so 73:3 should already include your changes, so... I'm not sure what's up.

Edit 2: Tested on both wordpress 5.2.1 (your version) and 5.3.2. Same issue in both for uploads.

**Comment by ZY-Ang** [comment]
Closing because very likely to be related to https://github.com/humanmade/S3-Uploads/issues/387

**Comment by keichan34** [comment]
@ZY-Ang Thanks for trying it out! Hm. It's been awhile, but I was able to upload media in my case. (I tried it out again, and it seems to work: https://dskhgdbzphjkm.cloudfront.net/sony-dsc/ )

I didn't see anything that jumped out at me in the console, and my lambda layer was created using the tools in the stackery repo.

Usual culprits for something like this would be permissions? But if you used the template.yaml, it should have taken care of that for you. Maybe network connectivity to the S3 service, if the Lambda is in a private subnet without a NAT gateway or an endpoint to S3?

I did have a lot of problems debugging stuff (like you said, the error logging is basically non-existent), which is why I didn't really continue maintaining this seriously.

**Comment by ZY-Ang** [comment]
@keichan34 Unfortunately, that's the first thing I checked and... It's definitely not an issue there or an issue with the VPC configuration.
- I am using the public subnets in my **default VPC** for ap-southeast-1 with the already pre-configured network.
- I tested the plugin (manual installation with composer) on EC2, and the CLI command `wp s3-uploads verify` work just fine. (turning on version history for s3 bucket shows they actually got uploaded then deleted, just that wp console uploads don't work)
- Also used admin-level IAM keys with `*` permissions but it still doesn't work.

It's likely an issue with the plugin that completely breaks new installations so I'm watching the issue there for now. I am not the only one experiencing the issue so I'll reupdate this again when there's a known resolution.
